pipeline/frontend/yaml/compiler/option.go

```diff
@@ -143,6 +143,17 @@ func WithProxy() Option {
 	)
 }
 
+// WithNetworks configures the compiler with additionnal networks
+// to be connected to build containers
+func WithNetworks(networks ...string) Option {
+	return func(compiler *Compiler) {
```

Comment:
let's model like this:

```
func WithNetworks(networks ...string) Option {
	return func(compiler *Compiler) {
		compiler.networks = networks
	}
}
```

the calling application will be responsible for sourcing the networks (from environment variable, or wherever)

```
opts.WithNetworks(
	os.Getenv("DRONE_NETWORKS"),
)
```

this would match how `WithVolumes` works. Check out 
https://github.com/cncd/pipeline/blob/master/pipec/compile.go#L277
https://github.com/cncd/pipeline/blob/master/pipeline/frontend/yaml/compiler/option.go#L28